### DISCUSSION TITLE
How to implement alternating row colors for a TreeDataGrid?

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#16638] [Q&A] [answered]
Since [TreeDataGrid](https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid) does not seem to have a discussion section I will try it here.

How can I implement alternating row colors for a TreeDataGrid?

**Comment by thevortexcloud** [comment]
You can probably do it with a selector. Something like

```xaml
<Style selector="TreeDataGridRow:nth-child(odd)">
<Setter Property="Background" Value="Purple"/>
</Style>
```

I have not tested that though, so it may need some tweaks.

https://docs.avaloniaui.net/docs/0.10.x/styling/selectors

**Reply by DevAM-Tools** [reply, marked as answer]
Thank you very much. You gave me the crucial clue.

```Xaml
<TreeDataGrid>
	<TreeDataGrid.Styles>
		<Style Selector="TreeDataGrid TreeDataGridRow:nth-child(2n)">
			<Setter Property="Background"  Value="#20808080"/>
		</Style>
	</TreeDataGrid.Styles>
</TreeDataGrid>
```

It could be found in the samples: https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/blob/d5d6496bc2a7d26daa2e7745c92d9a3d38cca531/samples/TreeDataGridDemo/MainWindow.axaml#L111
